lte/gateway/c/session_manager/test/test_cloud_reporter.cpp

```diff
@@ -65,14 +67,14 @@ class SessionReporterTest : public ::testing::Test {
   virtual void TearDown() {
     magma_service->Stop();
     reporter->stop();
+    delete evb;
   }
 
   // Timeout to not block test
   void set_timeout(uint32_t ms) {
-    std::thread([&]() {
+    std::thread([ms]() {
```

Comment:
per your comment, do you mean that capturing all elements by reference `[&]` may have been caused issues to the promise?

Reply:
@uri200  My hunch was that it is a bit suspicious to pass a stack defined parameter by reference into a thread. not 100% sure if it was causing any issues though. 
There's another `set_timeout` in `test_sessiond_integ` that has a similar issue. We define a promise/future on the stack in the main test function, and then pass a pointer to that to a separate thread.

Reply:
Something about this PR makes the tests less flaky, so I think I'm doing something right :D